rp2040-hal/src/multicore.rs

```diff
@@ -144,15 +144,20 @@ impl<'p> Core<'p> {
     }
 
     /// Spawn a function on this core.
+    ///
+    /// The closure should not return. It is currently defined as `-> ()` because `-> !` is not yet
+    /// stable.
+    ///
+    /// Core 1 will be reset from core 0 in order to spawn another task.
```

Comment:
```suggestion
    /// Core 1 will be reset from core 0 in order to spawn another task.
    ///
    /// Resetting a single core of a running program can have undesired consequences. Deadlocks are
    /// likely if the core being reset happens to be inside a critical section.
    /// It may even break safety assumptions of some unsafe code. So, be careful when calling this method
    /// more than once.
```